src/fs.rs

```diff
@@ -594,24 +593,21 @@ impl<S: 'static> StaticFiles<S> {
     pub fn with_pool<T: Into<PathBuf>>(dir: T, pool: CpuPool) -> StaticFiles<S> {
         let dir = dir.into();
 
-        let (dir, access) = match dir.canonicalize() {
+        let dir = match dir.canonicalize() {
             Ok(dir) => {
-                if dir.is_dir() {
-                    (dir, true)
-                } else {
+                if !dir.is_dir() {
                     warn!("Is not directory `{:?}`", dir);
```

Comment:
Now I think about it, maybe we should refactor it to return error to user instead of keeping it as it is

@fafhrd91 You wouldn't mind to change this API?

Reply:
@DoumanAsh sure, api changes are fine

Reply:
Do you want me to do that? Or should we leave this PR as-is?